### PR TITLE
Functionality to test examples programmatically

### PR DESCRIPTION
**Pull request by viktorsperling**
Introducing functionality for testing the setup scripts programmatically via the Travis CI.
This PR adds further jobs to the travis.yml in which chron jobs execute and test the setup scripts for the different supported frameworks with which Luigi can be used.
This is the cleaned version of the PR.

**Changes proposed in this pull request:**

- Adds additional jobs to travis.yml
- Adds new Cypress integration and config file

**Related issue(s)**
https://github.com/SAP/luigi/issues/1323

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: ~> 1.0 # enables build config validation beta feature: https://docs.travis-ci.com/user/build-config-validation
 language: node_js
 node_js:
-  - '12.18.4' #TODO: change it to 'node' once Travis supports the newest version
+  - '16.15.0' #TODO: change it to 'node' once Travis supports the newest version
```
Comment (wdoberschuetz) [inline .travis.yml:4]: (n2h) I guess you can remove the annotation

File: test/e2e-test-application/e2e/test3/0-setuptests/setup-test.spec.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  beforeEach(() => {
+    cy.visit(Cypress.env('url'));
+  });
+  describe('iFrame check', () => {
+    it('Check if iFrame exists in running application.', () => {
+      cy.wait(5000);
+      cy.get('iframe');
+    });
+  });
+});
```
Comment (wdoberschuetz) [inline test/e2e-test-application/e2e/test3/0-setuptests/setup-test.spec.js:11]: (q) If it was just a setup test for the development, remove it. If not, all good :).

Comment (viktorsperling) [reply test/e2e-test-application/e2e/test3/0-setuptests/setup-test.spec.js:11]: This is the required test for the process, described in the issue :).

Install example --> start webserver --> check if an iframe exists

**Comment by ndricimrr** [comment]
Apparently the last 3 commits do not seem to be running a Travis build. This happens sometimes, could you add an empty commit just to trigger it. Otherwise there's no way to see check how/if it works properly with the latest changes.

File: website/fiddle/package.json [hunk 3]
```diff
@@ -22,15 +22,15 @@
...
-    "@luigi-project/client": "1.22.0",
-    "@luigi-project/core": "1.22.0",
-    "@luigi-project/plugin-auth-oauth2": "1.22.0",
-    "@luigi-project/plugin-auth-oidc": "1.22.0",
-    "@sap-theming/theming-base-content": "~11.1.35",
+    "@luigi-project/client": "1.21.0",
+    "@luigi-project/core": "1.21.0",
+    "@luigi-project/plugin-auth-oauth2": "1.21.0",
+    "@luigi-project/plugin-auth-oidc": "1.21.0",
+    "@sap-theming/theming-base-content": "~11.1.28",
```
Comment (ndricimrr) [inline website/fiddle/package.json:29]: I can assume this was changed by mistake. please revert it. Then run npm install to revert the package-lock conflict

File: website/fiddle/package.json [hunk 4]
```diff
@@ -22,15 +22,15 @@
...
     "@ui5/webcomponents": "^1.3.0",
     "@ui5/webcomponents-fiori": "^1.3.0",
     "ace-builds": "1.4.14",
     "fs-extra": "^10.1.0",
-    "fundamental-styles": "0.23.0"
+    "fundamental-styles": "0.18.0"
```
Comment (ndricimrr) [inline website/fiddle/package.json:34]: please revert this change

File: website/fiddle/public/examples/microfrontends/fundamental/table-demo-page.html [hunk 5]
```diff
@@ -8,10 +8,6 @@
     <meta name="description" content="" />
     <meta name="keywords" content=" table demo page" />
     <title>Table Starter Page | SAP Fundamentals</title>
-    <link
```
Comment (ndricimrr) [inline website/fiddle/public/examples/microfrontends/fundamental/table-demo-page.html:11]: needs to be reverted as well

File: website/fiddle/public/examples/microfrontends/fundamental/tree-demo-page.html [hunk 6]
```diff
@@ -8,10 +8,6 @@
     <meta name="description" content="" />
     <meta name="keywords" content=" tree demo page" />
     <title>Tree Starter Page | SAP Fundamentals</title>
-    <link
```
Comment (ndricimrr) [inline website/fiddle/public/examples/microfrontends/fundamental/tree-demo-page.html:11]: here as well

File: core/src/core-api/config.js [hunk 7]
```diff
@@ -286,7 +286,7 @@ class LuigiConfig {
    * @memberof Configuration
    * @example
    * Luigi.clearNavigationCache();
-   * @since 1.19.0
+   * @since NEXT_RELEASE
```
Comment (ndricimrr) [inline core/src/core-api/config.js:289]: this and the related docu below [luigi-client-api.md](https://github.com/SAP/luigi/pull/2715/files#diff-4ff8ced4162e9cb3719d7544de4ec0ff185efbc3d3d655b610c857bb224e5f26), shouldn't be in this PR as well

**Review by ndricimrr — COMMENTED** [review]
There seems to be some merge conflicts and unrelated changes . Please make sure to revert all these changes

File: .travis.yml [hunk 8]
```diff
@@ -1,8 +1,10 @@
 version: ~> 1.0 # enables build config validation beta feature: https://docs.travis-ci.com/user/build-config-validation
 language: node_js
 node_js:
-  - '12.18.4' #TODO: change it to 'node' once Travis supports the newest version
+  - 14
```
Comment (ndricimrr) [inline .travis.yml:4]: you've added **14** and **16** and removed previous **12.18.4**. However, it seems though that you are only using 16. I would not add a new version if its not being used specifically. Alternatively, please leave '12.18.4' in there and node 16 can then be an alternative if Travis allows it.

**Review by ndricimrr — COMMENTED** [review]
A note on the node_js version:
The **node_js** version is used on our whole Travis CI, including e2e-tests, unit tests, checkmarx checks, npm publishing, documentation generation and all the rest. That's a lot of dedicated custom scripts. Changing the node version would introduce a breaking-change.
 If we need to upgrade the node_js version, we should do this on a dedicated other PR regarding node_js migration, where everything else needs to be throughly tested.

File: .travis.yml [hunk 9]
```diff
@@ -1,8 +1,10 @@
...
+  - 16
 dist: xenial
+script: echo "npm test temporarily disabled"
```
[the remark this answers is not in the document]
Comment (viktorsperling) [reply .travis.yml:7]: https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#default-build-script

Travis runs `npm test` on root level in its containers for every installed node version by default, this line disables this command.

File: website/fiddle/package-lock.json [hunk 10]
```diff
@@ -2432,9 +2432,9 @@
       "dev": true
     },
     "fundamental-styles": {
-      "version": "0.23.0",
-      "resolved": "https://registry.npmjs.org/fundamental-styles/-/fundamental-styles-0.23.0.tgz",
-      "integrity": "sha512-A5VtoDI8GebDziLmP4e1S4RXwlagglCgBUfHBCCe7OPzbIWH1s4rQ1mO6rp+Gd8aj0uvIe1mkHi+pNKEdqSg2Q=="
+      "version": "0.18.0",
```
Comment (ndricimrr) [inline website/fiddle/package-lock.json:2435]: this seems to be an old remaining change as well. 
Run npm install in this folder and commit the changes to fix this.

File: .travis.yml [hunk 11]
```diff
@@ -167,3 +168,50 @@ jobs:
       before_cache:
         - rm -rf ~/.npm/_logs
         - rm -rf ~/.npm/_cacache
+
+    # ----- SETUP EXAMPLES TESTS -----
+    - stage: 'Test Angular setup script'
```
Comment (ndricimrr) [inline .travis.yml:173]:
```suggestion
    - stage: 'Test Setup Scripts'
```

Comment (ndricimrr) [reply .travis.yml:173]: Commit this suggestion please, it can be misleading

**Review by ndricimrr — APPROVED** [review]
Looks good
